src/util/addEmojiSpace.js

```diff
@@ -0,0 +1,15 @@
+
+// regex pattern that checks if string starts with number of emoji followed by non-whitespace, non emoji character
+const emojiWithoutSpaceRegex = new RegExp(`^${emojiRegexString}+(?!(\\s|${emojiRegexString}))`, '')
+
+/**
+ * If a string starts with an emoji followed by a non-whitespace, non-emoji character,
+ * then insert a space between the emoji and the character.
+ */
+export const addEmojiSpace = text => {
+  const match = text.match(emojiWithoutSpaceRegex)
+  return (match ? match[0] + ' ' + text.substring(match[0].length) : text).trim()
```

Comment:
Let's remove the `trim` as it overreaches the purpose of `addEmojiSpace`.

If you are using it to remove a trailing space in the case where there is a match, instead detect that case and avoid adding the space to begin with, either via the regexp or checking the length of the rest of the text after the emoji.